### ISSUE TITLE
Get all metrics with LR find for all possible variations of models, 30 epochs of training, with early stopping of patience 5

### ISSUE THREAD

**Issue by dthiagarajan** [dthiagarajan/ptb-xl-research#3]
Using LR find in PyTorch Lightning:
- Train all model architectures for 30 epochs
- Use early stopping with patience of 5

**Problems**
* Distributed sync isn't happening right now in the epoch end methods - need to add that. Take advantage of the sync functions already in PL, e.g. [this](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/metrics/converters.py#L283)

* Doing LR find on distributed doesn't seem to be working too well. Might be easier to do something like

- Write a script dedicated to finding LRs (can probably be done on CPU with sufficient memory)
- Save JSON file to directory with those found LRs
- Use those as inputs for the training script by reading in when `find_lr` is specified as True
- Justify this in blog post (re: error when doing `auto_lr_find`/`find_lr`).

Potential fix: [here](https://github.com/PyTorchLightning/pytorch-lightning/pull/2790/files)

Try this potential fix on a smaller distributed compute node first, then scale up once it's confirmed.
